### ISSUE TITLE
Can't specify $queryRawUnsafe return type after extending prisma client

### ISSUE THREAD

**Issue by niba** [prisma/prisma#19862]
### Bug description

In prisma client you can pass generic to specify return type while executing queryRawUnsafe

```ts
  const data = await prisma.$queryRawUnsafe<{ id: number }[]>(
      SOME_SQL
   );
```

but when you decide to extend your prisma client then you can't do it anymore

```ts
  const extendedPrisma = prisma.$extends{{});
  const data = await extendedPrisma.$queryRawUnsafe<{ id: number }[]>( // ts error
      SOME_SQL
   );
```
error that I get:
` Type '{ id: number; }[]' does not satisfy the constraint '[query: string, ...values: any[]]'.
  Source provides no match for required element at position 0 in target `
this is a header of queryRawUnsafe
` (property) $queryRawUnsafe: <[query: string, ...values: any[]]>(query: string, ...values: any[]) => PrismaPromise_2<unknown>`

I use version with rewritten extension types => 4.16.0-dev.78

### How to reproduce

Steps above

### Prisma information

Code above

### Environment & setup

- OS: Windows
- Database: PostgreSQL
- Node.js version: 18

### Prisma Version

```
4.16.0-dev.78
```

**Comment by Jolg42** [comment]
@niba Thanks for the bug issue!

It's now published in our `patch-dev` version `4.16.1-dev.4`.

Let us know if this version fixes the issues, we're curious to get any feedback 🙌🏼 

Note: do not use in production, wait for the official `4.16.1` patch for that.

**Comment by luisrudge** [comment]
@Jolg42 fixed it for me! Thanks
